### ISSUE TITLE
Macro for allowing exception catching similar to EMSCRIPTEN_KEEPALIVE

### ISSUE THREAD

**Issue by ravisumit33** [emscripten-core/emscripten#15207]
In my code base, I have few functions which are in the `EXCEPTION_CATCHING_ALLOWED` list. Now, whenever I change something about these functions, their mangled name changes and I need to find those names and update them in the allow list. One option is to put them inside `extern 'C'` block but I don't want to do that as the code base is used for other platforms as well. I was wondering if there can be a macro similar to `EMSCRIPTEN_KEEPALIVE` that allows exception catching in the function?

**Comment by sbc100** [comment]
Unfortunately there is no way to do this today.   It would require some kind of custom attribute to be placed on the symbol, which is possible but would require upstream clang/llvm changes.

In the mean time are you using a separate file for maintaining your `EXCEPTION_CATCHING_ALLOWED` list?  You can specify it using `-sEXCEPTION_CATCHING_ALLOWED=@filename` and then list your symbols in a simple one-line-per-symbol format.

**Comment by ravisumit33** [comment]
@sbc100 Yes i am using a separate file to list symbols as use this file as `-sEXCEPTION_CATCHING_ALLOWED=@filename`. Changing mangled name of symbols is an issue.

**Comment by sbc100** [comment]
I think this would be great feature, but it does require some llvm hacking.   Perhaps we could piggy back on function annotations: https://github.com/WebAssembly/annotations since I imagine they will be adding the ability to add custom llvm attributes to symbols at the source level.

**Comment by ravisumit33** [comment]
Yes. The issue get much harder when the function you want to enable exception catching on is a template function. So there will be many symbols for that function and it is very hard to find all of those and list them in the exception catching allow list.
